lib/heap_sort.rb

```diff
@@ -1,8 +1,22 @@
-
+require_relative "min_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity: O(n log n), where n is the number of nodes and because this method adds the elements to a heap, then places them back in the array in order
+# Space Complexity: O(log n), due to recursive calls of the heap_up and heap_down methods within the add and remove methods
 def heapsort(list)
```

Comment:
👍   Since you are building a heap, the space complexity is O(n) (a heap of n elements).